src/xcode/ENA/ENA/Source/Services/HealthCertificateValidationService/Model/ValidationRulesCache.swift

```diff
@@ -0,0 +1,44 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import class CertLogic.Rule
+
+struct ValidationRulesCache: Codable {
+	
+	// MARK: - Init
+	
+	init(
+		validationRules: [Rule],
+		lastValidationRulesETag: String
+	) {
+		self.validationRules = validationRules
+		self.lastValidationRulesETag = lastValidationRulesETag
+	}
+	
+	// MARK: - Overrides
+	
+	// MARK: - Protocol Codable
+	
+	enum CodingKeys: String, CodingKey {
+		case validationRules
+		case lastValidationRulesETag
+	}
+	
+	init(from decoder: Decoder) throws {
+		let container = try decoder.container(keyedBy: CodingKeys.self)
+		
+		validationRules = try container.decode([Rule].self, forKey: .validationRules)
+		lastValidationRulesETag = try container.decode(String.self, forKey: .lastValidationRulesETag)
+	}
+	
+	// MARK: - Public
+	
+	// MARK: - Internal
+	
+	// MARK: - Private
+	
+	let validationRules: [Rule]?
+	let lastValidationRulesETag: String?
```

Comment:
In section private but no access rule set?

Reply:
1. They must be internal to be accessible in the service.
2. Right, I made them non-optional.